projects/observability/src/pages/explorer/explorer-dashboard-builder.ts

```diff
@@ -56,7 +57,261 @@ export class ExplorerDashboardBuilder {
     this.requestSubject.next(request);
   }
 
+  private getLayoutForElements(numberOfElements: number): object {
```

Comment:
Do you think we can automate this?
Also, if not automate, reuse the other parts apart from the `cell-span` thing?

Reply:
Added a function to generate column dimensions they're all the same. 

Not sure if one for row dimensions is possible since rows are unequal for less than 5 elements and equal afterwards.

For cell spans, can probably write a function to decrease lines of code but it'll get harder to read and modify code later. Should I add it?

Reply:
I think for rows, we are showing 4 items per row right? If that value is static, the row thing can be automated too. If you're disabling the option to add more values beyond a certain count, I am fine with things being static in code.

Reply:
Not always, here's the layout:
| No. of metrics | Grid used | Position of items |
| - | - | - |
| 1 | 3 x 3 | Single item in the middle cell, other cells empty |
| 2 to 4 | 3 x 4 | 4 items in the middle row, top & bottom row empty |
| 5 to 8 | 2 x 4 | 4 items each in two rows, full grid filled |

All layouts have different combination of rows & columns. Not sure what's the cap we should have for number of metrics. It's not a limitation really, I can keep adding more grid layouts but do we know any teams using panels with more than 8 metrics?